src/store/helpers.js

```diff
@@ -26,3 +28,111 @@ export function onlyHandleAPIError (error, handleFn) {
     throw error
   }
 }
+
+/**
+ * Defines a vues module that can be used to handle async request metadata.
+ *
+ * Instantiation:
+
+ const meta = defineRequestModule()
+ export const modules = { meta }
+
+ * Usage:
+
+async save ({ commit, dispatch }, group) {
+  dispatch('meta/request', {
+    id: `save-${group.id}`,
+    async run () {
+      const updatedGroup = await groups.save(group)
+      commit(types.RECEIVE_GROUP, { group: updatedGroup }) // other commits when successful
+    },
+  })
+},
+ */
+export function defineRequestModule () {
+  const types = {
+    REQUEST: 'Request',
+    RECEIVE_SUCCESS: 'Receive Success',
+    RECEIVE_ERROR: 'Receive Error',
+    CLEAR: 'Clear',
+  }
+
+  const state = {
+    entries: {},
+  }
+
+  const getters = {
+    get: state => id => state.entries[id] || ({
+      isWaiting: false,
+      error: null,
+      success: false,
+    }),
+    error: (state, getters) => id => field => {
+      const error = getters.get(id).error
```

Comment:
Nitpick: destructuring is nicer here :)

Reply:
Ah, good idea! Didn't spend much time thinking about this code, it just worked already ;)